runtime/ints.c

```diff
@@ -438,61 +438,52 @@ CAMLexport value caml_copy_int64(int64_t i)
   return res;
 }
 
-#define CAMLprim_int64_1(name)                                          \
-  CAMLprim int64_t caml_int64_##name##_native(int64_t);                 \
-                                                                        \
-  CAMLprim value caml_int64_##name(value v)                             \
-  { return caml_copy_int64(caml_int64_##name##_native(Int64_val(v))); } \
-                                                                        \
-  CAMLprim int64_t caml_int64_##name##_native
+CAMLprim value caml_int64_neg(value v)
+{ return caml_copy_int64(- Int64_val(v)); }
 
-#define CAMLprim_int64_2(name)                                          \
-  CAMLprim int64_t caml_int64_##name##_native(int64_t, int64_t);        \
-                                                                        \
-  CAMLprim value caml_int64_##name(value v1, value v2)                  \
-  { return caml_copy_int64(caml_int64_##name##_native(Int64_val(v1),    \
-                                                      Int64_val(v2))); } \
-                                                                        \
-  CAMLprim int64_t caml_int64_##name##_native
+CAMLprim value caml_int64_add(value v1, value v2)
+{ return caml_copy_int64(Int64_val(v1) + Int64_val(v2)); }
 
-CAMLprim_int64_1(neg)(int64_t i)
-{ return -i; }
+CAMLprim value caml_int64_sub(value v1, value v2)
+{ return caml_copy_int64(Int64_val(v1) - Int64_val(v2)); }
 
-CAMLprim_int64_2(add)(int64_t i1, int64_t i2)
-{ return i1 + i2; }
+CAMLprim value caml_int64_mul(value v1, value v2)
+{ return caml_copy_int64(Int64_val(v1) * Int64_val(v2)); }
 
-CAMLprim_int64_2(sub)(int64_t i1, int64_t i2)
-{ return i1 - i2; }
+#define Int64_min_int ((intnat) 1 << (sizeof(intnat) * 8 - 1))
 
-CAMLprim_int64_2(mul)(int64_t i1, int64_t i2)
-{ return i1 * i2; }
-
-CAMLprim_int64_2(div)(int64_t dividend, int64_t divisor)
+CAMLprim value caml_int64_div(value v1, value v2)
 {
+  int64_t dividend = Int64_val(v1);
+  int64_t divisor = Int64_val(v2);
   if (divisor == 0) caml_raise_zero_divide();
   /* PR#4740: on some processors, division crashes on overflow.
      Implement the same behavior as for type "int". */
-  if (dividend == ((int64_t)1 << 63) && divisor == -1) return dividend;
-  return dividend / divisor;
+  if (dividend == ((int64_t)1 << 63) && divisor == -1) return v1;
+  return caml_copy_int64(Int64_val(v1) / divisor);
```

Comment:
```suggestion
  return caml_copy_int64(dividend / divisor);
```
or have I missed something?

Reply:
I think you're right, but this is unrelated to this PR, is it?

Reply:
Actually, it is related to this PR!  I just manually reverted an old commit, not noticing that it introduced this sharing of `dividend`.  Will fix shortly.